File: 17/code_from_slides/calculator/calculator.cpp
#include <iostream>

int main(int argc, char* argv[]) {
    try {
        if(argc % 2 != 0) throw new std::runtime_error("");
        int accumulator = atoi(argv[1]);
        int index = 2;
        while(index+1 < argc) {
            int operand = atoi(argv[index+1]);
            switch((unsigned char)argv[index][0]) {
                case '+': accumulator += operand; break;
                case '-': accumulator -= operand; break;
                case 'x': accumulator *= operand; break;
                case '÷': accumulator /= operand; break;
                default: throw new 
                    std::runtime_error(std::string("Bad operator ") 
                                      + argv[index]);
            }
            index += 2;
        }
        std::cout << accumulator << std::endl;
    } catch(...) {
        std::cerr << "usage: " << argv[0] << " n1 [op n2]..." << std::endl;
    }
}

File: 17/code_from_slides/calculator/Calculator.java
public class Calculator {
    public static void main(String[] args) {
        try {
            if(args.length % 2 != 1) throw new 
                IllegalArgumentException(
                    "usage: java Calculator n1 [op n2]...");
            int accumulator = Integer.parseInt(args[0]);
            int index = 1;
            while(index+1 < args.length) {
                int operand = Integer.parseInt(args[index+1]);
                switch(args[index]) {
                    case "+" -> accumulator += operand;
                    case "-" -> accumulator -= operand;
                    case "x" -> accumulator *= operand;
                    case "÷" -> accumulator /= operand;
                    default -> throw new 
                        IllegalArgumentException("Bad operator " 
                                               + args[index]);
                }
                index += 2;
            }
            System.out.println(accumulator);
        } catch(Exception e) {
            e.printStackTrace();
            System.exit(-1);
        }
    }
}
